src/main/scala/mimir/algebra/function/TypeFunctions.scala

```diff
@@ -39,8 +39,19 @@ object TypeFunctions
       )
     }
 
+    for(functionName <- Seq("INTERVAL", "TO_INTERVAL")){
+      fr.register(
+        functionName,
+        (params: Seq[PrimitiveValue]) => 
+            { TextUtils.parseInterval(params.head.asString) },
+        _ match {
+          case Seq(_) => TDate()
+          case _ => throw new RAException("Invalid parameters to INTERVAL()")
+        }
+      )
+    }
```

Comment:
Suggest keeping this logic in [Cast](https://github.com/UBOdin/mimir/blob/master/src/main/scala/mimir/algebra/Cast.scala), which you've already done I see.